### ISSUE TITLE
Align vertical splitter below horizontal splitter

### ISSUE THREAD

**Issue by pongalongg** [ocornut/imgui#4261]
Hey,

`static float w = 200.0f;
			static float h = 300.0f;
			ImGui::PushStyleVar(ImGuiStyleVar_ItemSpacing, ImVec2(0, 0));
			ImGui::BeginChild("child1", ImVec2(w, h), true);
			ImGui::EndChild();
			ImGui::SameLine();
			ImGui::InvisibleButton("vsplitter", ImVec2(8.0f, h));
			if (ImGui::IsItemActive())
				w += ImGui::GetIO().MouseDelta.x;
			ImGui::SameLine();
			ImGui::BeginChild("child2", ImVec2(0, h), true);
			ImGui::EndChild();
			ImGui::InvisibleButton("hsplitter", ImVec2(-1, 8.0f));
			if (ImGui::IsItemActive())
				h += ImGui::GetIO().MouseDelta.y;
			ImGui::BeginChild("child3", ImVec2(0, 0), true);
			ImGui::EndChild();
			ImGui::PopStyleVar();`

I got this code from #125.
I'm trying to learn more about splitters and such.
The first thing i tried to do was move the vertical splitter to below the horizontal splitter. I've spent the last hour and a half on this, and for some reason, just can't seem to re-position it below the horizontal one. Any advice/help/recode would be much appreciated.

**Comment by ocornut** [comment]
Hello,
Would using Docking perhaps allow you to not have to use this splitter pattern?

**Comment by pongalongg** [comment]
My project is still back on 1.67 haha, i don't think i have docking. And i know a *lot* of stuff has changed since this version, for now i'd prefer to stay on 1.67.

So, i'd preferably like to use the current method, unless there's a more convenient method back in 1.67 you can think of?

**Comment by PathogenDavid** [comment]
> Is docking available on 1.67?

No, it's currently in [an experimental branch](https://github.com/ocornut/imgui/tree/docking), details in https://github.com/ocornut/imgui/issues/2109.

515ad6233550b417793f39cefbb5be0f80b94df9 would be the 1.67 version of the docking branch if you don't want to deal with upgrading stuff, but I'm not sure I'd recommend using such an old version of the docking branch.

> Furthermore, i'm looking to have my elements pre-placed. I don't want to have to drag them each time.

You can use the dock builder API to do this.

**Comment by pongalongg** [comment]
Thank you, i'll take your advice into consideration.

I'm going to focus on my original question for now, then learn about the docking branch. 

How am i able to position the vertical splitter *below* the horizontal?

**Comment by PathogenDavid** [comment]
The important thing is that the children above `hsplitter` must use `h` as their height and the children left of `vsplitter` must use `w` as their width.

You also have to use `ImGui::GetContentRegionAvail().y` for the height of `vsplitter` since its height since `h` represents the area above the `hsplitter` but you need the area below.

```cpp
ImGui::Begin("GH-4261 (Flipped)");
static float w = 200.0f;
static float h = 300.0f;
ImGui::PushStyleVar(ImGuiStyleVar_ItemSpacing, ImVec2(0, 0));

ImGui::BeginChild("child1", ImVec2(0, h), true);
ImGui::Text("Child1");
ImGui::EndChild();

ImGui::InvisibleButton("hsplitter", ImVec2(-1, 8.0f));
if (ImGui::IsItemActive())
    h += ImGui::GetIO().MouseDelta.y;

ImGui::BeginChild("child2", ImVec2(w, 0), true);
ImGui::Text("Child2");
ImGui::EndChild();

ImGui::SameLine();
ImGui::InvisibleButton("vsplitter", ImVec2(8.0f, ImGui::GetContentRegionAvail().y));
if (ImGui::IsItemActive())
    w += ImGui::GetIO().MouseDelta.x;
ImGui::SameLine();

ImGui::BeginChild("child3", ImVec2(0, 0), true);
ImGui::Text("Child3");
ImGui::EndChild();

ImGui::PopStyleVar();
ImGui::End();
```

![image](https://user-images.githubusercontent.com/278957/123328548-530d5b80-d501-11eb-97ad-60fccc05acdc.png)
